specifications/config.json

```diff
@@ -0,0 +1,127 @@
+{
+  "client": {
+    "apiKey":{
+      "file":"",
+      "id":"",
+      "secret":""
+    },
+    "cacheManager":{
+      "defaultTtl": 300,
+      "defaultTti": 300,
+      "caches": {
+        "account": {
+          "ttl": 300,
+          "tti": 300
+        }
+      }
+    },
+    "baseUrl": "https://api.stormpath.com/v1",
+    "connectionTimeout": 30,
+    "authenticationScheme": "SAUTHC1", // or "BASIC"
+    "proxy":{
+      "port": "",
+      "host": "",
+      "username": "",
+      "password": ""
+    }
+  },
+  "application": {
+    "name": "test",
+    "href": "https://..."
+  },
+  "web": {
+    "oauth2":{
+      "enabled": false,
+      "uri": "/oauth/token",
+      "password":{
+        "enabled": true,
+        "accessToken":{
+          "ttl": 3600
+        }
+      },
+      "client_credentials":{
+        "enabled": true,
+        "accessToken":{
+          "ttl": 3600
+        }
+      }
+    },
+    "accessTokenCookie":{
+      "name": "access_token",
+      "ttl": 3600,
+      "tti": 900,
+      "httpOnly": true,
+      "secure": null,
+      "path": "/",
+      "domain": null
+    },
+    "register": {
+      "enabled": false,
+      "uri": "/register",
+      "nextUri": "/",
+      "autoAuthorize": false,
+      "fields": [
```

Comment:
As an array, you'll have to re-define every element just to change one - which is really silly when you're trying to keep a small override file in your application.  For example, If I wanted to disable the surname field in my application I would have to redefine the entire array like this:

```json
{
  "fields": [
    {
      "name": "givenName",
      "required": true
    },
    {
      "name": "surname",
      "required": false   /* disabling the surname field */
    },
    {
      "name": "email",
      "required": true,
      "type": "email"
    },
    {
      "name": "password",
      "required": true,
      "type": "password"
    }
  ]
```

Whereas if we used a map, it's much more concise to simply disable a field like this:

```json
{
  "fields": {
    "surname": {
      "required": false
    }
  }
}
```
 
Also: comment's aren't allowed in JSON (as you'll see when this post is rendered)